functional_test.go

```diff
@@ -859,6 +864,62 @@ func TestGlobalRateLimits(t *testing.T) {
 	})
 }
 
+func TestGlobalRateLimitsWithLoadBalancing(t *testing.T) {
+	owner := cluster.PeerAt(2).GRPCAddress
+	peer := cluster.PeerAt(0).GRPCAddress
+	assert.NotEqual(t, owner, peer)
+
+	dialOpts := []grpc.DialOption{
+		grpc.WithResolvers(newStaticBuilder()),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"round_robin":{}}]}`),
+	}
+
+	address := fmt.Sprintf("static:///%s,%s", owner, peer)
+	conn, err := grpc.DialContext(context.Background(), address, dialOpts...)
+	require.NoError(t, err)
+
+	client := guber.NewV1Client(conn)
+
+	sendHit := func(status guber.Status, assertion func(resp *guber.RateLimitResp), i int) string {
+		ctx, cancel := context.WithTimeout(context.Background(), clock.Hour*5)
```

Comment:
this timeout is waaaay too high. Github actions will not run for more than 10 minutes by default

Reply:
just to be clear, this test wasnt ready to be merged as is, hence why it was in draft :) - it was simply to reproduce the culprit behaviour that was reported in the bug.